### PR TITLE
Preconfigure and document GitHub OAuth login

### PR DESCRIPTION
**Pull request by sbrunner**
GSGMF-1867

### CODE REVIEW

File: doc/integrator/c2cwsgiutils.rst [hunk 1]
```diff
@@ -1,15 +1,49 @@
...
+ ``c2cwsgiutils`` is a framework assisting in development, integration and administration of WSGI applications.
+``c2cgeoportal`` uses ``c2cwsgiutils``.
 See its `documentation <https://github.com/camptocamp/c2cwsgiutils/#camptocamp-wsgi-utilities>`__.
 
 The entry point URL is ``<application main URL>/c2c``, where you will have a dashboard with the
-c2cwsgiutils and c2cgeoportal services.
+``c2cwsgiutils`` and ``c2cgeoportal`` services.
 
-To use the protected tools, you should at least set ``C2C_SECRET`` to a secret password in the ``env.project`` file.
+The ``invalidate`` view is also protected with this sec
```
Comment (arnaud-morvan) [inline doc/integrator/c2cwsgiutils.rst:13]:
```suggestion
The ``invalidate`` view is also protected with this secret.
```
But we do not know anymore what is "this secret".
Seems that you should remove completely this sentence here.

File: doc/integrator/tilegeneration.rst [hunk 2]
```diff
@@ -136,8 +136,8 @@ Web Interface
 ~~~~~~~~~~~~~
 
 It is possible to run tile generation commands through a web interface located at URL
-``<application main URL>/tiles/admin/``. This interface is protected by a password that
-must be specified in the ``C2C_SECRET`` environment variable in the ``env.project`` file.
+``<application main URL>/tiles/admin/``. This interface is protected by a password
```
Comment (arnaud-morvan) [inline doc/integrator/tilegeneration.rst:139]:
```suggestion
``<application main URL>/tiles/admin/``. This interface require an authentication
```

**Review by arnaud-morvan — APPROVED** [review]
LGTM

File: doc/integrator/c2cwsgiutils.rst [hunk 3]
```diff
@@ -1,15 +1,49 @@
...
 
-The ``invalidate`` view is also protected with this secret.
+.. _integrator_c2cwsgiutils_auth:
+
+Authentication
+--------------
+
+To have basic authentication of those tools (and on tile generation) yon can set
```
Comment (ochriste) [inline doc/integrator/c2cwsgiutils.rst:20]: yon > you

File: doc/integrator/c2cwsgiutils.rst [hunk 4]
```diff
@@ -1,15 +1,49 @@
...
+``C2C_SECRET`` in the ``env.project`` file to have single password without brute force protection.
+
+You can also activate the authentication on GitHub with OAuth2, the configuration is in the
+``env.project`` file:
+
+* Verify that the ``C2C_AUTH_GITHUB_REPOSITORY`` is correctly configured.
+* By default, the ``C2C_AUTH_GITHUB_ACCESS_TYPE`` is set to admin then only admins users of the previously
+  defined repository have access.
+* You should ask Camptocamp for a new secret and set it in ``C2C_AUTH_GITHUB_CLIENT_SECRET``.
+  I will be defined in the GitHub organizations settings, on 'Developer settings', 'OAuth Apps',
```
Comment (ochriste) [inline doc/integrator/c2cwsgiutils.rst:30]: I > It

File: doc/integrator/c2cwsgiutils.rst [hunk 5]
```diff
@@ -1,15 +1,49 @@
...
+  'Geoservicies` application, 'Generate a new client secret',
+  the URL should also be configured in the redirect proxy.
+* Defined the ``C2C_AUTH_GITHUB_SECRET`` randomly.
```
Comment (ochriste) [inline doc/integrator/c2cwsgiutils.rst:33]: Defined > Define

File: doc/integrator/c2cwsgiutils.rst [hunk 6]
```diff
@@ -1,15 +1,49 @@
...
+If you want to be independent you should:
+
+* Verify that the ``C2C_AUTH_GITHUB_REPOSITORY`` is correctly configured.
+* By default, the ``C2C_AUTH_GITHUB_ACCESS_TYPE`` is set to admin then only admins users of the previously
+  defined repository have access.
+* Create an OAuth Apps on GitHub: GitHub organizations settings, on 'Developer settings', 'OAuth Apps',
+  'New OAuth App`, fill the form, reopen the application, 'Generate a new client secret',
+  ``C2C_AUTH_GITHUB_CLIENT_ID`` and ``C2C_AUTH_GITHUB_CLIENT_SECRET`` should be configured with the
+  obtain information.
+* Defined the ``C2C_AUTH_GITHUB_SECRET`` randomly.
```
Comment (ochriste) [inline doc/integrator/c2cwsgiutils.rst:45]: Defined > Define

File: doc/integrator/c2cwsgiutils.rst [hunk 7]
```diff
@@ -1,15 +1,49 @@
...
+
+
+For more information see the
```
Comment (ochriste) [inline doc/integrator/c2cwsgiutils.rst:48]: remove "the" or add "documentation page" at the end

File: doc/integrator/c2cwsgiutils.rst (hunk 1 above)
Comment (sbrunner) [reply doc/integrator/c2cwsgiutils.rst:13]: =>
The ``invalidate`` view is protected with the same authentication, see bellow.

**Pull request merged**